### PR TITLE
Generalize note for `with-meta` (guide, Step A)

### PR DESCRIPTION
**Pull request by dfuenzalida**
The bullet point for `with-meta` describes how to implement it for Mal functions, but it's applicable to all Mal types, while the last sentence can be reordered to emphasize the details specific to metadata on functions.

### CODE REVIEW

File: process/guide.md [hunk 1]
```diff
@@ -1648,11 +1648,11 @@ implementation.
   * `meta`: this takes a single mal function argument and returns the
     value of the metadata attribute.
   * `with-meta`: this function takes two arguments. The first argument
-    is a mal function and the second argument is another mal
-    value/type to set as metadata. A copy of the mal function is
+    is a mal value and the second argument is another mal
+    value/type to set as metadata. A copy of the mal type is
```
Comment (kanaka) [inline process/guide.md:1652]: This should really be "value" since it's a copy of the value and not a copy of the value's type.

**Review by kanaka — CHANGES_REQUESTED** [review]
@dfuenzalida Thanks for the catch! One small tweak I've requested and then I'll merge it.

File: process/guide.md (hunk 1 above)
Comment (dfuenzalida) [reply process/guide.md]: Sorry for the delay in getting back to this, I'll update the PR

**Comment by kanaka** [comment]
@dfuenzalida Apologies for the very slow response. I'm finally getting around to looking at tickets and PRs again. This change is good. I rebased your changes onto the current HEAD and squashed the commits together here: https://github.com/kanaka/mal/commit/a1f1a72d2ba9e4e368fff3df5ee6e65963b3945b

**Pull request closed without merging**